### DISCUSSION TITLE
Ability to set font when updating form fields

### DISCUSSION THREAD

**Discussion by rsk2** [py-pdf/pypdf#2770] [Q&A] [answered]
## Explanation

Is there a way to set the font, font size, and font weight when writing to PDF form fields?

**Comment by stefan6419846** [comment]
This has been added in 4.3.0: https://pypdf.readthedocs.io/en/stable/modules/PdfWriter.html#pypdf.PdfWriter.update_page_form_field_values (PR #2636). AFAIK the font weight usually is a font property and thus not supported there, unless you reference a regular, bold and/or italic font.

**Comment by Butterfly** [comment]
Just a lurker here chiming in to say that the ability to change fonts is a decades old top level accessibility/usability talking point. So basically.. Thank you for thinking of the needs of others! :purple_heart:

**Comment by pubpub-zz** [comment]
Just to clarify, the fonts needs to exists within the PDF. We are not yet able to import a TTF file and use it directly.

**Reply by rsk2** [reply]
Ok. So how does the font id work? How do I know which font ids exist in the PDF? I tried randomly using `/F1`, but I got this error: `Font dictionary for /F1 not found.`

**Comment by pubpub-zz** [comment]
@rsk2 
can you have a look at the tests folder to see how to use it ?

**Reply by rsk2** [reply]
@pubpub-zz I looked at the tests folder; maybe it will be better if I share some code. This is the code I am using to update the PDF:

`writer = PdfWriter(clone_from="mypdf.pdf")`
`writer.update_page_form_field_values(None, {"some_key": ("Text_2", "/F1", 0)})`

But I get the error: `Font dictionary for /F1 not found`. Same issue occurs with `/F2` and `/F3`

It does work if I don't provide a font as follows:

` writer.update_page_form_field_values(None, {"some_key": ("Text_2", "", 0)})`

Also, the font size input doesn't seem to make a difference. If I pass in a size like 1 or 10, the output is the same as that for 0 (autosize).

**Reply by pubpub-zz** [reply]
@rsk2  can you provide the pdf you are trying to fill up

**Reply by rsk2** [reply]
Sorry, I had to remove the PDF I uploaded previously due to some issues. But here's another PDF in which the same issues occur: [Stop Payment Form.pdf](https://github.com/user-attachments/files/16488169/Stop.Payment.Form.pdf)

**Reply by abubelinha** [reply]
> Also, the font size input doesn't seem to make a difference. If I pass in a size like 1 or 10, the output is the same as that for 0 (autosize).

I am having the same issue (3rd value in the tuple doesn't have any effect in changing the font size in filled form answers).
Can anyone provide a simple working example of this functionality?
Thanks!

**Reply by abubelinha** [reply]
> I am having the same issue (3rd value in the tuple doesn't have any effect in changing the font size in filled form answers). 

@rsk2 in my case the problem was solved: I was not passing the `auto_regenerate=False` as shown in the example
https://github.com/py-pdf/pypdf/blob/main/tests/test_writer.py#L2263-L2266

When adding `auto_regenerate=False`, passed font sizes (3rd element in tuples) are now being properly applied.

**Comment by pubpub-zz** [comment, marked as answer]
when you look within the document you can see that 3 fonts are available:
![image](https://github.com/user-attachments/assets/d4f372e0-8d39-4892-8717-a83de3df6c2a)

make a new try with "/HeBo", "/Helv", "/ZaDb"

**Reply by oreanroy** [reply]
what is this tool you are using to the structure of the pdf file?

**Reply by stefan6419846** [reply]
This is the Apache PDFBox Debugger standalone tool.
